example/main.ml

```diff
@@ -0,0 +1,45 @@
+open Rtree
+open Printf
+(* Sample Point of Interest data structure *)
+type poi = {
+  name : string;
+  latitude : float;
+  longitude : float;
+}
+
+
+module Rtree =  struct
+
+(* Function to insert POIs into the R-tree *)
+let insert_pois rtree pois =
+  List.iter (fun poi -> Rtree.insert rtree (poi.latitude, poi.longitude, poi)) pois
+
+(* Function to load geospatial data from a CSV file *)
+let load_geospatial_data_from_csv filename =
+  let ic = open_in filename in
+  let csv = Csv.of_channel ~has_header:true ic in
+  let pois = ref [] in
+  Csv.iter
```

Comment:
To make this more _functional_ I think we should use [fold_left](https://ocaml.org/p/csv/latest/doc/Csv/index.html#val-fold_left) instead of iter + mutable state :))

Reply:
Thanks @patricoferris, will do that